app/src/main/java/com/a5corp/weather/GlobalActivity.java

```diff
@@ -35,15 +43,48 @@ protected void onResume() {
         }
 
         super.onResume();
-        Intent intent;
+        intent=new Intent();
 
         if (prefs.getLaunched()) {
-            intent = new Intent(GlobalActivity.this, WeatherActivity.class);
+            if(prefs.getCity()!=null && prefs.getCity().length()>0){
+                FetchWeather fw=new FetchWeather(this);
+                try {
+                    Info json=fw.execute(prefs.getCity()).get();
+                    intent = new Intent(GlobalActivity.this, WeatherActivity.class);
+                    intent.putExtra("json",json);
+                } catch (ExecutionException e) {
+                    e.printStackTrace();
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }else{
+                FetchWeather fw=new FetchWeather(this);
+                try {
+                    float lat=prefs.getLatitude();
+                    float lon=prefs.getLongitude();
+                    Info json=fw.execute(String.valueOf(lat),String.valueOf(lon)).get();
+                    intent = new Intent(GlobalActivity.this, WeatherActivity.class);
+                    intent.putExtra("json",json);
```

Comment:
Create a new variable called `SPLASH_DATA` inside `Constants.java` and assign it the value `splash_data`.

Use that as the first parameter of `putExtra()`. Do the same thing in the `WeatherActivity` and `WeatherFragment` wherever you have used the string "json" for passing data.

Reply:
If I assign splash_data to a static variable everytime it's generated then I think there is no need of sending the data through bundle as whenever data is required we can access it via class name.
When user changes the city via entering a new one on fab then json data regarding that city has be loaded in update_weather function, then that time I can assign null to SPLASH_DATA as it won't be required further until the app is launched next time and global_activity is loaded.
What do you say?

Reply:
Line 67, you have `intent.putExtra("json",json);` . What I meant to say is, in the first parameter, you have "json" as the key name. This should be a constant, which is why I told you to place `splash_data` in `Constants.java`

Reply:
Ok got that!